test/e2e/network/loadbalancer.go

```diff
@@ -670,6 +670,35 @@ var _ = common.SIGDescribe("LoadBalancers", func() {
 		if framework.ProviderIs("azure") {
 			ginkgo.By("switching back to interal type LoadBalancer, with static IP specified.")
 			internalStaticIP := "10.240.11.11"
+
+			// For a cluster created with CAPZ, node-subnet may not be "10.240.0.0/16", e.g. "10.1.0.0/16".
+			getInternalStaticIPInSubnet := func() string {
+				nodes, err := cs.CoreV1().Nodes().List(context.TODO(), metav1.ListOptions{})
+				if err != nil {
+					framework.Logf("error listing Nodes, err: %v", err)
+					return ""
+				}
+				for _, node := range nodes.Items {
+					_, isControlPlane := node.Labels["node-role.kubernetes.io/master"]
+					if isControlPlane {
+						continue
+					}
+					for _, addr := range node.Status.Addresses {
+						if addr.Type == v1.NodeInternalIP {
+							splitted := strings.Split(addr.Address, ".")
+							if len(splitted) == 4 {
+								return fmt.Sprintf("%s.%s.11.11", splitted[0], splitted[1])
+							}
+						}
+					}
+				}
+				return ""
+			}
```

Comment:
there are already helpers in `test/e2e/framework/node/resource.go` , please double check but I think that you can 

```
node,err := e2enode.GetRandomReadySchedulableNode()
internalIP, err := e2enode.GetInternalIP(node)
```

Reply:
Yes, but actually for azure, the control-plane doesn't have the same subnet as workers. So I created a new method to pick a worker node.

Reply:
that-'s ok ... I vaguely remember some method  doing exactly the same , but I can't find it ...  :smile: